src/browser/api/channel.ts

```diff
@@ -177,18 +181,18 @@ export module Channel {
 
     export function sendChannelMessage(identity: Identity, payload: any, messageId: number, ack: AckFunc, nack: NackFunc): void {
         const { uuid, name, payload: messagePayload, action: channelAction, providerIdentity } = payload;
-        const targetIdentity = { uuid, name };
+        const intendedTargetIdentity = { uuid, name };
 
         const ackToSender = createAckToSender(identity, messageId, providerIdentity);
 
-        sendToIdentity(targetIdentity, {
+        sendToIdentity(intendedTargetIdentity, {
             action: CHANNEL_APP_ACTION,
             payload: {
                 ackToSender,
                 providerIdentity,
                 action: channelAction,
                 senderIdentity: identity,
-                payload: messagePayload
+                payload: {intendedTargetIdentity, ...messagePayload}
             }
```

Comment:
I thought we were going to put this property one level up?  Weren't we saying messagePayload could be a string?

Reply:
This commit was before our convo. Been testing it since, about to push up changes.